packages/gatsby/src/utils/worker/child/render-html.ts

```diff
@@ -175,13 +192,38 @@ export const renderHTMLProd = async ({
           path: pagePath,
           unsafeBuiltinsUsageByPagePath,
         }
-        throw e
+
+        // If we're in Preview-mode, write out a simple error html file.
+        if (isPreview) {
+          const pageData = await readPageData(publicDir, pagePath)
+          const truncatedPageData = truncateObjStrings(pageData)
+
+          const html = `<h1>Preview build error</h1>
+        <p>There was an error when building the preview page for this page ("${pagePath}").</p>
+        <h3>Error</h3>
+        <pre><code>${e.stack}</code></pre>
+        <h3>Page component id</h3>
+        <p><code>${pageData.componentChunkName}</code></p>
+        <h3>Page data</h3>
+        <pre><code>${JSON.stringify(truncatedPageData, null, 4)}</code></pre>`
```

Comment:
can we pull this out into a function that builds the preview build error html elsewhere for readability? It'll make it easier to iterate on it later on as well

Reply:
Hmm this is pretty short code right now. It'll be easy to pull out into a seperate function in the future if that proves helpful.